opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/aggregation/dsl/MetricAggregationBuilder.java

```diff
@@ -194,6 +204,27 @@ private Pair<AggregationBuilder, MetricParser> make(CardinalityAggregationBuilde
     return Pair.of(aggregationBuilder, parser);
   }
 
+  /**
+   * Make {@link TopHitsAggregationBuilder} for take aggregations.
+   */
+  private Pair<AggregationBuilder, MetricParser> make(TopHitsAggregationBuilder builder,
+                                                      Expression expression,
+                                                      Expression size,
+                                                      Expression condition,
+                                                      String name,
+                                                      MetricParser parser) {
+    String fieldName = ((ReferenceExpression) expression).getAttr();
+    builder.fetchSource(fieldName, null);
+    builder.size(size.valueOf(null).integerValue());
```

Comment:
i saw a lot of place use xxx.valueOf(null) seems Expression should define `ExprValue valueOf()` interface, and only LiteralExpression has implementation.

Reply:
we can create the issue and make the change in following PR.

Reply:
https://github.com/opensearch-project/sql/issues/976